### PR TITLE
Fixes #188

### LINKED ISSUE #188: SmidgeLinkTagHelper overrides script preload to CSS
```html
<link rel="preload" as="script" href="@bundleName" />
```

This gets incorrectly rewritten to a CSS path.

### PR DESCRIPTION
**Pull request by PhyxionNL**

### CODE REVIEW

File: src/Smidge/TagHelpers/SmidgeLinkTagHelper.cs [hunk 1]
```diff
@@ -1,47 +1,36 @@
...
-        private const string HrefIncludeAttributeName = "asp-href-include";
-        private const string HrefExcludeAttributeName = "asp-href-exclude";
-        private const string FallbackHrefAttributeName = "asp-fallback-href";
-        private const string SuppressFallbackIntegrityAttributeName = "asp-suppress-fallback-integrity";
-        private const string FallbackHrefIncludeAttributeName = "asp-fallback-href-include";
-        private const string FallbackHrefExcludeAttributeName = "asp-fallback-href-exclude";
-        private const string FallbackTestClassAttributeName = "asp-fallback-test-class";
-        private const string FallbackTestPropertyAttributeName = "asp-fallback-test-property";
-        private const string FallbackTestValueAttributeName = "asp-fallback-test-value";
-        private const string AppendVersionAttributeName = "asp-append-version";
```
Comment (PhyxionNL) [inline src/Smidge/TagHelpers/SmidgeLinkTagHelper.cs:26]: Moved all of these inline as there's no other usage of them.

File: src/Smidge/TagHelpers/SmidgeLinkTagHelper.cs [hunk 2]
```diff
@@ -83,29 +72,32 @@ public override async Task ProcessAsync(TagHelperContext context, TagHelperOutpu
...
+            if (context.AllAttributes.TryGetAttribute("as", out TagHelperAttribute attribute) && attribute.Value is not "style")
             {
-                throw new InvalidOperationException("Smidge tag helpers do not support the ASP.NET tag helpers: " + string.Join(", ", _invalid));
+                return;
             }
 
-            var result = (await _smidgeHelper.GenerateCssUrlsAsync(Source, Debug)).ToArray();
-            var currAttr = output.Attributes.ToDictionary(x => x.Name, x => x.Value);
-            using (var writer = new StringWriter())
+            var attributes = output.Attributes.ToDictionary(x => x.Name, x => x.Value);
```
Comment (PhyxionNL) [inline src/Smidge/TagHelpers/SmidgeLinkTagHelper.cs:85]: Removed useless array allocation and use async overloads.

File: src/Smidge/TagHelpers/SmidgeLinkTagHelper.cs [hunk 3]
```diff
@@ -83,29 +72,32 @@ public override async Task ProcessAsync(TagHelperContext context, TagHelperOutpu
...
-            if (context.AllAttributes.Any(x => _invalid.Contains(x.Name)))
+            if (context.AllAttributes.Any(x => _invalidAttributes.Contains(x.Name)))
+            {
+                return;
+            }
+
+            if (context.AllAttributes.TryGetAttribute("as", out TagHelperAttribute attribute) && attribute.Value is not "style")
             {
-                throw new InvalidOperationException("Smidge tag helpers do not support the ASP.NET tag helpers: " + string.Join(", ", _invalid));
+                return;
```
Comment (PhyxionNL) [inline src/Smidge/TagHelpers/SmidgeLinkTagHelper.cs:82]: Fixes #188 by checking if there's an "as" attribute defined with a value other than "style".

**Comment by PhyxionNL** [comment]
@Shazwazza GitHub Packages fails, probably access token expired:
```warn : Your request could not be authenticated by the GitHub Packages service. Please ensure your access token is valid and has the appropriate scopes configured.```

**Comment by Shazwazza** [comment]
@PhyxionNL awesome! Yeah can probably remove the GH packages integration, its never used.

**Pull request merged**